### PR TITLE
Adjust config files to work with 'console-dev.{cluster}...' domain

### PR DESCRIPTION
**Pull request by parostatkiem**
**Description**
I've made some changes in config files to make the Console work with `console-dev.{cluster}.cluster.kyma.cx` instead of `console.yfactory.local`.

#### Reasons of this change:
- (as @y-kkamil discovered) cluster's DEX refused to authorise unknown domain. `console-dev.{cluster_addr}`,  is a known domain for it, `yfactory.local` is not :)
- why would we use `console.yfactory.local`? It was weird.

Changes proposed in this pull request:

- change `setClusterConfig.sh` script 
- change `config.js` and `luigi-config` in Console to respect "the new way" of setting domains

### How to test it
1. Pull changes from this PR
2. create `setClusterConfig.sh` file the _kyma-project_ folder and paste the below code into it.
3. In your terminal, go to _kyma-project_ folder and execute `./setClusterConfig local` or `./setClusterConfig {cluster_name}` (the usage is the same as it was in the previous version of this script) or `./setClusterConfig {cluster_URL}`
4. Run `npm run bootstrap` in Console's root folder, then `npm start`
5. Check if everything works, then switch to another cluster/local kyma by repeating points **3 and 4**

In case you entered {cluster_name} as a param, the script will use your `CLUSTER_HOST` **environment** variable for creating a full cluster URL.

``` batchfile
#!/bin/zsh
 
SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
HOST="$(echo $CLUSTER_HOST)"

if [  -z "$HOST"  ]
  then
    echo "\e[91mIt looks like your \e[92mCLUSTER_HOST\e[91m environment variable is not set.
    The script might not work properly.
    \e[39m"
  else
    echo "Cluster host read from your environment: \e[92m$HOST\e[91m"
fi
 
if [ $1 = "local" ]
  then
    DOMAIN=kyma.local
  else
    if [[ $1 == *"."* ]]; then 
        DOMAIN=$1 # given argument (cluster name) contains a dot => it is a full cluster URL
    else
        DOMAIN=$1.$HOST # given argument (cluster name) doesn't contain a dot => it is just a cluster name
    fi

fi

LOCALDOMAIN=console-dev.$DOMAIN

echo "\e[39mSetting config for: \e[36m$1\e[0m"
echo ""

# replace variables in config.js. Running 'npm run bootstrap' is necessary to bring them deeper into Console
sed -i '' "s/var localDomain = .*/var localDomain = '$LOCALDOMAIN';/" $SCRIPTPATH/console/config.js
sed -i '' "s/var domain = .*/var domain = '$DOMAIN';/" $SCRIPTPATH/console/config.js


echo "Root permissions needed to remove previous cluster->localhost bindings in /etc/hosts"
sudo sed -i '' "/.$HOST/d" /etc/hosts


# add new cluster->localhost binding to hosts file
echo "127.0.0.1 console-dev.$DOMAIN localhost"| sudo tee -a /etc/hosts


echo "Added ClusterConfig to Luigi and Console"
echo ""
echo -e "Please run \e[91mnpm run bootstrap\e[0m in a root Console folder"
echo -e "Then you can open \e[93mhttp://console-dev.$DOMAIN:4200\e[0m"
echo ""
echo ""
echo "Luigi OIDC Settings which you can use for extendedConfig.js:"
echo ""
echo "use: \"openIdConnect\","
echo "openIdConnect: {"
echo "  authority: \"https://dex.$DOMAIN\"," 
echo "  client_id: \"console\","
echo "  scope: \"audience:server:client_id:kyma-client audience:server:client_id:console openid profile email groups\","

exit 0
```

Note that you can put either _cluster name_ or _cluster URL_ as a param. The script automatically detects your choice and adds `.cluster.stage.faros.kyma.cx` to make URL from your _cluster name_.
**Related issue(s)**

### CODE REVIEW

File: core/src/app/app.config.ts [hunk 1]
```diff
@@ -35,19 +39,19 @@ const config = {
...
-  serviceCatalogModuleUrl: `https://catalog.${domain}`,
-  serviceInstancesModuleUrl: `https://instances.${domain}`,
-  serviceBrokersModuleUrl: `https://brokers.${domain}`,
-  docsModuleUrl: `https://docs.${domain}`,
-  logsModuleUrl: `https://log-ui.${domain}`,
+  serviceCatalogModuleUrl: `https://catalog.${localDomain}`,
+  serviceInstancesModuleUrl: `https://instances.${localDomain}`,
+  serviceBrokersModuleUrl: `https://brokers.${localDomain}`,
+  docsModuleUrl: `https://docs.${localDomain}`,
+  logsModuleUrl: `https://log-ui.${localDomain}`,
```
Comment (maxmarkus) [inline core/src/app/app.config.ts]: If we change all those to `localDomain` will it then still have the right domain for production?

Comment (parostatkiem) [reply core/src/app/app.config.ts]: Good question, I'll check it.

Comment (parostatkiem) [reply core/src/app/app.config.ts]: For now, our Cluster doesn't work due to external reasons but so far I was able to test this PR on local cluster Console (the one that comes with local Kyma installation) and it works 🍾 
I'll test it on real cluster ASAP 😃

**Comment by parostatkiem** [comment]
Blocked until we have working remote cluster to test it (to be sure it works not only locally).

**Comment by parostatkiem** [comment]
@maxmarkus I switched Console image to this PR on Swinka cluster and it works. Is there any other way to _check if it still works on production_?
image

**Comment by sjanota** [comment]
I was able to check it with https://console.hasselhof-cluster.cluster.stage.faros.kyma.cx and it was fine

**Comment by parostatkiem** [comment]
@y-kkamil 
1. Yes, it is needed. Try to comment this line, erase your `hosts`, **not** install kyma and run Console with remote Kyma cluster. You won't be able to enter Console from the browser until you manually add this line to `hosts`.
2. Hmm, it works for me.
3. I adjusted it to remove a line containing  environment's CLUSTER_HOST variable. I don't thing there's a possibility for the script to "know" which cluster addresses have been used in the history 🤷‍♂

**Pull request merged**